model-engine/model_engine_server/api/llms_v1.py

```diff
@@ -172,6 +176,9 @@ async def list_model_endpoints(
     """
     Lists the LLM model endpoints owned by the current owner, plus all public_inference LLMs.
     """
+    external_interfaces.monitoring_metrics_gateway.emit_route_call_metric(
```

Comment:
Main comment is if there is a way to do "automatically" do this without having to copy-pasta on each route and to automatically used the defined route name. 

I have looked into something like this before, and I think we have to put it into a dependency. We can't do it in the middleware because we need the auth dep in order to get the user_id.

Reply:
https://stackoverflow.com/a/66634433

If we want to add user auth info as a tag, I think our hand is forced to add these obs additions as a dependency (as opposed to middleware)